chat.revolt.RevoltDesktop.yaml

```diff
@@ -36,30 +33,32 @@ modules:
 - name: revolt
   buildsystem: simple
   build-options:
-    append-path: /usr/lib/sdk/node16/bin
+    append-path: /usr/lib/sdk/node18/bin
     env:
-      npm_config_nodedir: /usr/lib/sdk/node16
-  build-commands:
-  - cp -a revolt-desktop-1.0.6 /app/main
-  - desktop-file-edit --set-icon=$FLATPAK_ID revolt-desktop.desktop
-  - install -Dm644 revolt-desktop.desktop /app/share/applications/$FLATPAK_ID.desktop
-  - install -Dm644 build/icons/icon.png /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
-  - install -Dm644 ../$FLATPAK_ID.metainfo.xml /app/share/metainfo/$FLATPAK_ID.metainfo.xml
-  - install -Dm755 ../revolt-desktop.sh /app/bin/revolt-desktop
-  subdir: main
+      npm_config_nodedir: /usr/lib/sdk/node18
   sources:
-  - type: archive
-    url: https://github.com/revoltchat/desktop/archive/v1.0.6.tar.gz
-    sha256: 84c06e3924e53d3272f7c2be9832a259f710c75a7a36d2732c3cb7a0e119e366
-    dest: main
+
   - type: archive
     url: https://github.com/revoltchat/desktop/releases/download/v1.0.6/revolt-desktop-1.0.6.tar.gz
     sha256: eba79090ca83fa0d549cb3df062a44c76b940cdd96aec9d7b7823a1399f694ad
-    dest: main
     strip-components: 0
   - type: script
     dest-filename: revolt-desktop.sh
     commands:
-    - zypak-wrapper.sh /app/main/revolt-desktop "$@"
+    - zypak-wrapper.sh /app/bin/revolt-desktop-1.0.6/revolt-desktop "$@"
```

Comment:
Maybe this versioned dir could be avoided altogether? It would make it later possible to rely on flatpak-external-data-checker for updating the sources.

One idea I have in mind is to unpack everything into a directory (something like what was done with directory main) and using `strip-components: 1` (this is also the default value so that option could be removed). That way the files are unpacked into a directory with a generic name (no version number).

Reply:
i see i will try that

Reply:
i fixed it